### PR TITLE
[10.0][MIG] account_invoice_mass_sending (from 12.0)

### PR DESCRIPTION
**Pull request by RobinetDenisAcsone**
Backport of https://github.com/OCA/account-invoicing/pull/518

### CODE REVIEW

File: account_invoice_mass_sending/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.addons.queue_job.job import job
+from odoo.exceptions import UserError
+
+JOB_QUEUE_CHANNEL = "root.PREPARE_SEND_PRINT_INVOICE"
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/models/account_invoice.py:9]: Hi @RobinetDenisAcsone ,

In V10, only the "send by email" feature is assumed by the final wizard method invoked (i.e. `send_mail_action()`).

Thus, IMO, everywhere the term "Print" is used as a part of a name or a label (intended to the final user), il should be deleted.

File: account_invoice_mass_sending/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        for rec in self:
+            rec.with_delay().do_send_print()
+
+    @api.multi
+    @job(default_channel=SEND_QUEUE_CHANNEL)
+    def do_send_print(self):
+        for rec in self:
+            if not rec.partner_id.email:
+                raise UserError(_(
+                    "Missing email address on customer " "'{customer_name}'."
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/models/account_invoice.py]: Is the break in this string into 2 substrings useful ?

File: account_invoice_mass_sending/i18n/fr.po [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#: code:addons/account_invoice_mass_sending/models/account_invoice.py:30
+#, python-format
+msgid "The sending of {in_progress_invoices_count} invoices is already in progress."
+msgstr "L'envoi de {in_progress_invoices_count} factures est déjà en cours."
+
+#. module: account_invoice_mass_sending
+#: code:addons/account_invoice_mass_sending/models/account_invoice.py:36
+#, python-format
+msgid "The sending of {invoices_count} invoices will be processed in background."
+msgstr "L’envoi de {invoices_count} sera traité en arrière-plan."
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/i18n/fr.po]: ... factures ...

File: account_invoice_mass_sending/i18n/fr.po [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+"Last-Translator: <>\n"
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: account_invoice_mass_sending
+#: model:ir.model.fields,help:account_invoice_mass_sending.field_account_invoice_sending_in_progress
+msgid "If checked, the invoice is already being processed, and it will prevent the sending of a duplicated mail."
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/i18n/fr.po:20]: Does not appear in the .pot

**Review by Olivier-LAURENT — CHANGES_REQUESTED** [review]
Rmk: it's more of a backport than a migration.

**Comment by RobinetDenisAcsone** [comment]
@Olivier-LAURENT changes done

**Comment by Olivier-LAURENT** [comment]
Hi @ThomasBinsfeld 

An opinion about all these methods and xmlids nammed with the term "print" in this V10 version ? Thx

**Comment by ThomasBinsfeld** [comment]
> Hi @ThomasBinsfeld
> 
> An opinion about all these methods and xmlids nammed with the term "print" in this V10 version ? Thx

@Olivier-LAURENT @RobinetDenisAcsone I don't think it's a big deal since we won't start from this v10 version for future migrations of the module. I would let as it because keeping the code closer to the orignal (v12) makes backports of patches and database migration easier.

File: account_invoice_mass_sending/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,3 @@
+On the invoices list view, select the invoices to send and click on 'Action > Send & print'.
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/readme/USAGE.rst]: s/Send & print/Send/

File: account_invoice_mass_sending/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+
+It will use the existing 'SEND & PRINT' action on the invoice with the default value.
```
Comment (Olivier-LAURENT) [inline account_invoice_mass_sending/readme/USAGE.rst]: s/SEND & PRINT/SEND/

**Review by Olivier-LAURENT — APPROVED** [review]
Just 2 remaining minor comments. Thx @RobinetDenisAcsone .

**Pull request closed without merging**